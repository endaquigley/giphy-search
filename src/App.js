import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import GalleryContainer from './GalleryContainer';
import Gallery from './Gallery';
import Pagination from './Pagination';

const Header = styled.header`
  padding: 20px;
  text-align: center;
`;

const Content = styled.main`
  background-color: var(--secondary-colour);
`;

const Footer = styled.footer`
  color: white;
  padding: 20px;
`;

const Heading = styled.h1`
  font-size: 50px;
  line-height: 1;
  margin-top: 0;
  margin-bottom: 15px;
`;

class App extends Component {

  state = {
    page: 0,
    query: 'dog'
  }

  updateQuery(query) {
    this.setState({
      page: 0,
      query: query
    });
  }

  prevPage() {
    this.setState(({ page }) => {
      return { page: page - 1 }
    });
  }

  nextPage() {
    this.setState(({ page }) => {
      return { page: page + 1 }
    });
  }

  render() {
    return (
      <Fragment>

        <Header>
          <Heading>
            { this.state.query === 'dog' ? '🐶' : '🐱' }
          </Heading>

          <button onClick={ () => this.updateQuery('dog') } disabled={ this.state.query === 'dog' }>Dog Search</button>
          <button onClick={ () => this.updateQuery('cat') } disabled={ this.state.query === 'cat' }>Cat Search</button>
        </Header>

        <Content>
          <GalleryContainer page={ this.state.page } query={ this.state.query } render={({ data }) => (
            <Gallery data={ data } />
          )}/>
        </Content>

        <Footer>
          <Pagination
            page={ this.state.page }
            prevPage={ () => this.prevPage() }
            nextPage={ () => this.nextPage() }
          />
        </Footer>

      </Fragment>
    );
  }

}

export default App;
