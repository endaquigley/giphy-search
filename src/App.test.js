import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

const response = {
  data: [
    {
      id: 1,
      images: {
        preview: { mp4: 'https://enda.ie/filename-01.mp4' },
        original: { mp4: 'https://enda.ie/filename-01.mp4' }
      }
    },
    {
      id: 2,
      images: {
        preview: { mp4: 'https://enda.ie/filename-01.mp4' },
        original: { mp4: 'https://enda.ie/filename-01.mp4' }
      }
    }
  ]
};

global.fetch = jest.fn().mockImplementation(() => {
  return new Promise((resolve, reject) => {
    resolve({ json: () => response });
  });
});

it('renders without crashing', () => {
  const component = document.createElement('div');
  ReactDOM.render(<App />, component);

  expect(component).toMatchSnapshot();
});
