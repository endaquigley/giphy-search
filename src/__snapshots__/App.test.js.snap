// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <header
    class="sc-htoDjs ggvsSU"
  >
    <h1
      class="sc-gZMcBi hHyiFP"
    >
      üê∂
    </h1>
    <button
      disabled=""
    >
      Dog Search
    </button>
    <button>
      Cat Search
    </button>
  </header>
  <main
    class="sc-dnqmqq bxGFOx"
  >
    <div
      class="sc-bZQynM jgZutv"
    />
  </main>
  <footer
    class="sc-iwsKbI fHEDOg"
  >
    <div
      class="sc-gzVnrw bYgKLD"
    >
      <button
        disabled=""
      >
        Prev Page
      </button>
      <span>
        Page 
        1
      </span>
      <button>
        Next Page
      </button>
    </div>
  </footer>
</div>
`;
